packages/jsapi-utils/src/SessionUtils.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  ConnectOptions,
+  CoreClient,
+  IdeConnection,
+  IdeSession,
+} from '@deephaven/jsapi-types';
+import {
+  requestParentResponse,
+  SESSION_DETAILS_REQUEST,
+} from '@deephaven/jsapi-utils';
+import Log from '@deephaven/log';
+import shortid from 'shortid';
+import NoConsolesError, { isNoConsolesError } from './NoConsolesError';
+
+const log = Log.module('SessionUtils');
+
+export interface SessionConfig {
+  type: string;
+  id: string;
+}
+
+export interface SessionDetails {
+  workerName?: string;
+  processInfoId?: string;
+}
+
+export interface SessionWrapper {
+  session: IdeSession;
+  connection: IdeConnection;
+  config: SessionConfig;
+  details?: SessionDetails;
+}
+
+/**
+ * @returns New connection to the server
+ */
+export function createConnection(websocketUrl: string): IdeConnection {
+  log.info(`Starting connection to '${websocketUrl}'...`);
+
+  return new dh.IdeConnection(websocketUrl);
+}
+
+/**
+ * Create a new session using the default URL
+ * @returns A session and config that is ready to use
+ */
+export async function createSessionWrapper(
+  connection: IdeConnection,
+  details: SessionDetails
+): Promise<SessionWrapper> {
+  log.info('Getting console types...');
+
+  const types = await connection.getConsoleTypes();
+
+  log.info('Available types:', types);
+
+  if (types.length === 0) {
```

Comment:
Should log available types after throwing for none available so there's not a double message in error cases